src/components/GenericDropdown.tsx

```diff
@@ -15,38 +14,90 @@ const DropdownButton = styled.button`
 	border: 1px, solid, black;
 `;
 
+const DropdownMenu = styled.ul`
+	list-style: none;
+	background-color: ${({ theme }) => theme.colors.main[0]};
+	text-align: start;
+	z-index: 2;
+	position: absolute;
+	border: 1px solid rgba(0, 0, 0, 0.04);
+	box-shadow: 0 16px 24px 2px rgba(0, 0, 0, 0.14);
+	padding: 0;
+	margin: 0;
+	border-radius: 10px;
+`;
+
 const Container = styled.div`
 	display: flex;
 	flex-direction: column;
 	height: 100%;
 	width: 100%;
 `;
 
-function GenericDropdown({ buttonDisplay, dropdownItems }: Props) {
+export const ListLink = styled(Link)`
+	display: block;
+	height: 100%;
+	text-decoration: none;
+	color: ${({ theme }) => theme.colors.text[2]};
+	padding: 1em;
+	&:hover {
+		color: ${({ theme }) => theme.colors.primary[0]};
+	}
+`;
+
+export const ListButton = styled.div`
+	display: block;
+	height: 100%;
+	text-decoration: none;
+	color: ${({ theme }) => theme.colors.text[2]};
+	padding: 1em;
+	&:hover {
+		color: ${({ theme }) => theme.colors.primary[0]};
+	}
+`;
+
+function GenericDropdown({
+	buttonDisplay,
+	children,
+}: React.PropsWithChildren<Props>) {
 	const ref = useRef<HTMLDivElement>(null);
 	const buttonRef = useRef<HTMLButtonElement>(null);
-	const { isComponentVisible, setIsComponentVisible } = useComponentVisible(
+	const [componentVisible, setComponentVisible] = useComponentVisible(
 		ref,
 		false
 	);
 	const handleClick = () => {
-		setIsComponentVisible(!isComponentVisible);
+		setComponentVisible(!componentVisible);
 	};
 
+	const [listTopPosition, setListTopPosition] = useState(-9999);
+
+	useEffect(() => {
+		const getMenuButtonBottom = () => {
+			if (buttonRef.current != null) {
```

Comment:
What does `buttonRef.current != null` mean? Could we pull this into a variable?

Reply:
I mean this seems like overkill haha, but I extracted `buttonRef.current` (DOM element) as `menuButton` .

```typescript
const getMenuButtonBottom = () => {
	const menuButton = buttonRef.current;
	if (menuButton != null) {
		return menuButton.getBoundingClientRect().bottom;
	}
	return HIDE_LIST;
};
```
And at this point I think it is very clear, without giving the return type a name.

Reply:
I don't think `buttonRef.current` is what's unclear, rather `buttonRef.current != null`, is this checking the existence of the button? or if the button is enabled vs. disabled?

I'd rather see something like
```js
const buttonExists = buttonRef.current !+ null;

if (buttonExists) {
```

Reply:
I disagree on that one, because I think `menuButton != null` reads very clearly as menuButton is defined.